### ISSUE TITLE
Errors with a nested controller

### ISSUE THREAD

**Issue by boazin** [mharris717/ember-cli-pagination#83]
Here's my scenario:

I have 2 models, session and test.
session has_many tests.
Because the tests aren't embedded in the JSON, I use RSVP to fetch it.

So routes/session.js is:
```
import Ember from 'ember';
import RouteMixin from 'ember-cli-pagination/remote/route-mixin';

export default Ember.Route.extend(RouteMixin, {
  model: function(params) {
    params.paramMapping = {page: "page",
      perPage: "page_size",
      total_pages: "total_num_pages"};
    return Ember.RSVP.hash({
      session: this.get('store').find('session', params.session_id),
      tests: this.findPaged('test',params)
    });
  },

  setupController: function(controller, model) {
    controller.set('model', model.session);
    var testsController = this.controllerFor('tests');
    testsController.set('model', model.tests);
  }
});
```

When I break inside setupController I see all the data correctly (total_pages etc.)
But when rendering it somehow gets lost
(getting ```no currentPage for page-numbers``` and more warnings...)

is it related to the fact that in the model it is has_many?

Another piece of info:
in session.hbs I call:
```
{{render "tests" tests}}
```

And the actual page-numbers is in - tests.hbs, and the relevant data is in controllers/tests.js:
```
  queryParams: ["page","perPage"],
  page: 1,
  perPage: 10,

  pageBinding: "content.page",
  perPageBinding: "content.perPage",
  totalPagesBinding: "content.totalPages",
```

Any ideas to investigate? (tried for some time alone, no success)
I appreciate all the hard work you do on this package, it is awesome...

**Comment by boazin** [comment]
Just to shed some more light - I tried to enable pagination on the "tests" route itself - and it works perfectly, so it definitely has something to do with the nesting.

Any assistance will be appreciated...

**Comment by boazin** [comment]
Don't know why it solved it (would still love for an explanation) but doing ```{{render "tests"}}``` solved the issue.

From the docs:
```
(When no model is provided it gets the singleton instance of the corresponding controller
When a model is provided it gets a unique instance of the corresponding controller)
```

**Comment by mharris717** [comment]
So here's my reading:

When you call this in `session.hbs`

```javascript
{{render "tests" tests}}
```

The `tests` property on your session controller is not set to anything. You've (arguably correctly) set the model of the TestsController to the tests array. So this call is passing null as the model, while the call not specifying a model uses the singleton TestsController on which you've already set the model. 

Funny sidenote: I wrote the original version of that page in the docs, but the part you quoted was added after, I'm not sure it's correct, and it conflicts with something a couple paragraphs down. It's my understanding that it's always the singleton controller, although that may have changed and I could be wrong.
